src/builder/cfg.rs

```diff
@@ -95,6 +97,20 @@ impl<B: AsMut<Hugr> + AsRef<Hugr>> CFGBuilder<B> {
             inputs: Some(input),
         })
     }
+    pub(crate) fn from_existing(base: B, cfg_node: Node) -> Result<Self, BuildError> {
```

Comment:
It's not pub - but yes, docstring added.

I've clued about the entry node (the thing is that creating a CFGBuilder *doesn't* create an entry node - you have to do that explicitly later using e.g. `simple_entry_builder` and it will prepend that onto the list of children rather than append).

And added a test. (That creates a disconnected basic block, but it'll do for the test.)